### DISCUSSION TITLE
I would like to know how to change the Image Source in codebehind.

### DISCUSSION THREAD

**Discussion by SID9-HoneyBee** [dotnetprojects/SVGImage#53] [Q&A]
I am using this library to use SVG as background for InkCanvas.
It's working very well.

I load multiple SVG files at once
I want to replace an image based on user input.

To do that, it seems that you have to create an SVG image with the absolute file path in the backend and apply it every time.

But I don't know how to apply this library.

May I know how to apply this as an image when I open a specific SVG file with OpenFileDialog ?

Did I express my intentions correctly?
Could you please review the method using the library as well?

ImageSource = svg:SvgIcon ...

Is this correct?
```cs
 private async void btn_OpenFile_Click(object sender, RoutedEventArgs e)
        {
            openFileDialog = new OpenFileDialog();
            //openFileDialog.Filter = _OpenFileFilter;

            if (openFileDialog.ShowDialog().Value == true)
            {
                string filePath = openFileDialog.FileName;

                string fileName = System.IO.Path.GetFileNameWithoutExtension(filePath);

                SVGImage.SVG.SVGImage image = new SVGImage.SVG.SVGImage();

                image.UriSource = new Uri(filePath, UriKind.Absolute);

                inkCanvasImage.ImageSource = (ImageSource)image.ImageSource;

            }
        }
```
```xml
<InkCanvas x:Name="inkCanvas" EditingMode="None"
                   Grid.Row="0">
            <InkCanvas.Background>
                <ImageBrush x:Name="inkCanvasImage"
                            
                            ImageSource="{svg:SvgIcon Source=1.svg}"
                            Stretch="Uniform">

                </ImageBrush>
            </InkCanvas.Background>
            <InkCanvas.RenderTransform>
                <TransformGroup>
                    <ScaleTransform x:Name="inkCanvasScale"/>
                    <TranslateTransform x:Name="rectTrans"/>
                </TransformGroup>
            </InkCanvas.RenderTransform>
        </InkCanvas>
```

**Comment by paulushub** [comment]
> Is this correct?

"Correct" may not be the right word here, since this will work but not what you might need to do.
`SvgIcon` is a markup extension not a control, so the `Image` control will activate this whenever it needs an `ImageSource`.
However, `SVGImage` is a control, you do not want to create a control just to get an `ImageSource`, that is why I added the
`SvgIcon` and `SvgIconConverter`.

**Why SvgIconConverter?**
WPF markup extensions have to extend the abstract class `MarkupExtension` (there is no defined interface) which does not derive from `DependencyObject` (like most WPF classes), so you cannot do WPF binding on markup extensions.
So to be able to support the binding, `SvgIconConverter` also implements `IValueConverter` to make it  a WPF converter based on markup extension, which also makes the XAML syntax different.

See the sample below for demonstration of the `SvgIconConverter`; illustrating binding to resources, local files, web files and list view. For your use case, see the third tab **PageOther**.
[IconConverterSample.zip](https://github.com/dotnetprojects/SVGImage/files/6671937/IconConverterSample.zip)

**Reply by SID9-HoneyBee** [reply]
Thank you for your kind reply.

I have a few additional questions
May I proceed?
